airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/test/commands.py

```diff
@@ -73,10 +105,20 @@ async def test(pipeline_context: ClickPipelineContext):
             ),
         )
         .with_workdir(f"/airbyte/{poetry_package_path}")
-        .with_exec(["poetry", "install"])
+        .with_exec(["poetry", "install", "--with=dev"])
         .with_unix_socket("/var/run/docker.sock", dagger_client.host().unix_socket("/var/run/docker.sock"))
         .with_env_variable("CI", str(pipeline_context.params["is_ci"]))
-        .with_exec(["poetry", "run", "pytest", test_directory])
+        .with_workdir(f"/airbyte/{poetry_package_path}")
     )
 
-    await pytest_container
+    soon_command_executions_results = []
+    async with asyncer.create_task_group() as poetry_commands_task_group:
+        for command in commands_to_run:
+            logger.info(f"Running command: {command}")
+            soon_command_execution_result = poetry_commands_task_group.soonify(run_poetry_command)(test_container, command)
+            soon_command_executions_results.append(soon_command_execution_result)
+
+    for result in soon_command_executions_results:
+        stdout, stderr = result.value
+        logger.info(stdout)
+        logger.error(stderr)
```

Comment:
Do we still get the exit on failure behavior that we want? I know in format, when running multiple steps concurrently we had to add explicit error exiting

Reply:
If an execution fails the async context will throw an `ExceptionGroup` which will lead to a click command failure and a failed CI pipeline.

Reply:
Ah, great! I guess before it was different because we wanted to continue all the others and collect successes and failures? I suppose that could be helpful here too, but it's not blocking for me, as its all our code that we're checking

Reply:
Yes you're right, we could a Dagger exec error handling to better handle success / failure.